elastic/kibana
# [Monitoring] Fixes bug creating default rules when opening dropdown

Resolves https://github.com/elastic/kibana/issues/107935

move rules creation out of closing popover which is called anytime the popover goes away.

### Test
- open the "Alerts and rules" dropdown and close it.
- verify either by waiting for the popup to NOT show up that rules were created or look at the kibana log to see no default rules were created.  
- open the dropdown again and click "create default rules".  see in the kibana log or toast that default rules were created.
